core/templates/pages/contributor-dashboard-page/services/contribution-and-review.service.spec.ts

```diff
@@ -13,96 +13,381 @@
 // limitations under the License.
 
 /**
- * @fileoverview Tests that the contribution-and-review service is working as
- * expected.
+ * @fileoverview Unit tests for contribution and review service
  */
 
-// TODO(#7222): Remove the following block of unnnecessary imports once
-// UserService.ts is upgraded to Angular 8.
-import { UpgradedServices } from 'services/UpgradedServices';
-// ^^^ This block is to be removed.
-
-require(
-  'pages/contributor-dashboard-page/services/' +
-  'contribution-and-review.service.ts');
-
-describe('ContributionAndReviewService', function() {
-  var ContributionAndReviewService, $httpBackend;
-  var suggestion1;
-  var opportunityDict1;
-  var mockSuggestionsBackendObject;
-  var expectedSuggestionDict;
-  var onSuccess;
-
-  beforeEach(angular.mock.module('oppia'));
-  beforeEach(angular.mock.module('oppia', function($provide) {
-    var ugs = new UpgradedServices();
-    for (const [key, value] of Object.entries(ugs.getUpgradedServices())) {
-      $provide.value(key, value);
-    }
-  }));
-
-  beforeEach(angular.mock.inject(function($injector) {
-    ContributionAndReviewService = $injector.get(
-      'ContributionAndReviewService');
-    $httpBackend = $injector.get('$httpBackend');
-
-    suggestion1 = {
-      suggestion_id: 'suggestion_id_1',
-      target_id: 'skill_id_1',
-    };
-    opportunityDict1 = {
-      skill_id: 'skill_id_1',
-      skill_description: 'skill_description_1',
-    };
-    mockSuggestionsBackendObject = {
-      suggestions: [
-        suggestion1
-      ],
-      target_id_to_opportunity_dict: {
-        skill_id_1: opportunityDict1,
-      },
-    };
-    expectedSuggestionDict = {
-      suggestion: suggestion1,
-      details: opportunityDict1,
-    };
-  }));
+import { TestBed, fakeAsync, tick } from '@angular/core/testing';
+import { HttpClientTestingModule } from '@angular/common/http/testing';
+import { ContributionAndReviewService } from './contribution-and-review.service';
+import { UrlInterpolationService } from 'domain/utilities/url-interpolation.service';
+import { ContributionAndReviewBackendApiService }
+  from './contribution-and-review-backend-api.service';
+import { SuggestionBackendDict } from 'domain/suggestion/suggestion.model';
+
+describe('Contribution and review service', () => {
+  let cars: ContributionAndReviewService;
+  let carbas: ContributionAndReviewBackendApiService;
+
+  let suggestion1 = {
```

Comment:
Can we convert all the `let`s that are note being reassigned to `const`?

Reply:
will be cover under PR #14471.